Userland/Utilities/realsize.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2024, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/CharacterTypes.h>
+#include <AK/Optional.h>
+#include <AK/StringView.h>
+#include <AK/Types.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    TRY(Core::System::pledge("stdio rpath"));
+
+    StringView argument;
+
+    Core::ArgsParser args_parser;
+    args_parser.set_general_help(
+        "Show the 'real' size of a number with a suffix (possibly).");
+    args_parser.add_positional_argument(argument, "Number with possibly a suffix", "number");
+    args_parser.parse(arguments);
+
+    if (argument.is_null() || argument.is_empty())
+        return Error::from_string_literal("Invalid value");
+
+    unsigned suffix_multiplier = 1;
+    auto suffix = argument[argument.length() - 1];
+
+    switch (suffix) {
+    case 'k':
+        suffix_multiplier = KiB;
+        argument = argument.substring_view(0, argument.length() - 1);
+        break;
+    case 'm':
+        suffix_multiplier = MiB;
+        argument = argument.substring_view(0, argument.length() - 1);
+        break;
+    case 'g':
+        suffix_multiplier = GiB;
+        argument = argument.substring_view(0, argument.length() - 1);
+        break;
+    case 'K':
+        suffix_multiplier = KB;
+        argument = argument.substring_view(0, argument.length() - 1);
+        break;
+    case 'M':
+        suffix_multiplier = MB;
+        argument = argument.substring_view(0, argument.length() - 1);
+        break;
+    case 'G':
+        suffix_multiplier = GB;
+        argument = argument.substring_view(0, argument.length() - 1);
+        break;
+    }
+
+    Optional<unsigned> numeric_optional = argument.to_number<unsigned>();
+    if (!numeric_optional.has_value())
+        return Error::from_string_literal("Invalid size-value");
+
+    auto numeric_value = numeric_optional.value() * suffix_multiplier;
+    if (numeric_value < 1)
+        return Error::from_string_literal("Invalid size-value");
```

Comment:
If someone wants to create a 0KB file, are we really the ones who should be keeping them from doing that?